codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +148,109 @@ private static void insert(Node root, int symbol, int code, byte length) {
             current.children[i] = terminal;
         }
     }
+
+    private final class Decoder {
+        private final DecoderProcessor processor = new DecoderProcessor();
+        public byte[] decode(ByteBuf buffer, int length) throws IOException {
+            try {
+                buffer.forEachByte(buffer.readerIndex(), length, processor);
+                buffer.skipBytes(length);
+                return processor.end();
+            } finally {
+                processor.reset();
+            }
+        }
+
+        private final class DecoderProcessor implements ByteProcessor {
+            private final ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
in this case we actually want a `byte[]` because that is the underlying storage for the `Http2Headers`. If we use a `ByteBuf` the underlying storage may not be `byte[]`. I'm fine with `ByteArrayOutputStream` in this case.